src/main/java/org/example/App.java

```diff
@@ -26,5 +21,15 @@ public static void main( String[] args ) {
                 .limit(randomPointsGenerator.getNumPoints())
                 .collect(Collectors.toList());
         System.out.println(points);
+
+        //Observer logic
+        User ourMentor = new User(null, "Andrei", "Trukhanovich", "atrukhanovich@solvd.com");
+        DesiredPath routing = new DesiredPath();
+        routing.subscribe(ourMentor);
+        ourMentor.setDesiredPath(routing);
+        // calculate the optimal route
+        Route route = new Route();
+        routing.setRoute(route);
```

Comment:
Setting the route for the routing here doesn't update the routing's route in the user since you set ourMentor's desired path to an empty desired path before setting the desired path's route. Can either set the route to routing before setting routing to desired path or see if you can update within the set methods.

Reply:
Done I think. Not 100% clear what you mean.